components/base/sensorcontrolled/sensorcontrolled.go

```diff
@@ -195,10 +241,24 @@ func (sb *sensorBase) SetVelocity(
 	var sensorCtx context.Context
 	sensorCtx, sb.sensorLoopDone = context.WithTimeout(context.Background(), timeOut)
 
+	if err := sb.angularVelocityLoop.Start(); err != nil {
```

Comment:
why so many  starts?

Reply:
If I have a loop running in the background in this driver - my reading of the control loop code is that it will interfere with other API commands that are called independently. I do not want the control loop to be "active" and interfering with `MoveStraight`, `Spin`, or `SetPower` if linear and angular velocity sensors exist and are being polled.

I can set a flag and return immediately from the `SetState` function I suppose - like so:
```go
type endpointBase struct {
          ....
          active bool
          ....
}

func (sb *sensorBase) setPolling(...){
         ....
         sb.endpointBase.active = false
         .....
}


func (endpointBase) SetState(...) error {
         if !endpointBase.active {
         // return immediately do not set the state of the base.
         return nil
         }
         
         .....
         endpointBase.SetVelocity(...)
}

```

Or something that follows this code.

Reply:
The issue is - there's already a polling loop in spin - one that can be converted to its loop as well - the endpoint for that would be a SetVelocity command and a State that polls an orientation sensor, with some math applied to the orientation signal to figure out if we're at target or overshot after crossing the first-fourth quadrant boundary. That logic is not in the `sensorbase/spin.go` file.